service/handlers.go

```diff
@@ -113,3 +126,47 @@ func (e *ExecuteStruct) GetClient(projectID string) (client.RancherClient, error
 	}
 	return *apiClient, nil
 }
+
+func saveWebhook(webhookRequestData map[string]interface{}) error {
+	config := config.GetConfig()
+	if _, ok := webhookRequestData["name"].(string); !ok {
+		return fmt.Errorf("Name not found after decode")
+	}
+	webhookName := webhookRequestData["name"].(string)
+	apiClient, err := client.NewRancherClient(&client.ClientOpts{
```

Comment:
The client needs to be scoped to the user. You should pass in the same client that you later pass into the driver. Don't create it here.